pkg/apis/v1alpha1/dnsrecord_types.go

```diff
@@ -107,7 +107,7 @@ type DNSRecordSpec struct {
 	ManagedZoneRef *ManagedZoneReference `json:"managedZone,omitempty"`
 	// +kubebuilder:validation:MinItems=1
 	// +optional
-	Endpoints []*Endpoint `json:"endpoints"`
+	Endpoints []*Endpoint `json:"endpoints,omitempty"`
```

Comment:
Out of interest, why was this needed? I have no issue with it, other than it probably doesn't make sense with that validations checking for 1 item. We either allow it to be empty or we don't.

Reply:
I faced a nil dereference at one point when this was empty. That was at a point where I hadn't implemented the use case of including all endpoints when the checks resulted in there being no endpoints present. So I think it's unlikely that we'd hit that now but i kept in in case that case came up again in order to prevent any crashes in the code.

Reply:
We should probably remove https://github.com/Kuadrant/multicluster-gateway-controller/pull/412/files#diff-2c474b44d62073e226cd94ece69ad23db20801a324cec059d85f1a9128acebd0L108. Can be done later.